### PR TITLE
Split crates

### PR DESCRIPTION
**Pull request by DaniPopes**
As described in #78, this PR moves crates into the `crates/` directory, the binaries to `svm-rs/src/bin/`, and adds the `cli` and `solc` features to gate their dependencies.
These are:
- `anyhow`
- `clap`
- `indicatif`
- `itertools`
- `dialoguer`
- `console`
- `tokio` ("full" feature)

After doing this, I found that the `rand`, `tokio` and `tempfile` dependencies were only used with `#[cfg(test)]`, so I moved them into `dev-dependencies`.

I also updated the README and the badge links.

#### This reduces `svm-rs` crate size from [~30kB](https://crates.io/crates/svm-rs) to ~23.6KiB, and library dependencies from 183 to 108.

### CODE REVIEW

**Comment by roynalnaruto** [comment]
Thank you for these changes. I will merge once the CI is successful :)

File: crates/svm-builds/build.rs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#![allow(dead_code)]
```
Comment (roynalnaruto) [inline crates/svm-builds/build.rs:1]: Why this change?

**Review by roynalnaruto — COMMENTED** [review]
Given the fact that there are several changes in the `Cargo.toml` files, I would prefer using a regular expression for allowing patch updates instead of specifying a patch.

So something like `^1.0.0` instead of `1.0.17`. Apart from this nitpick, this PR overall looks good. Again, thank you so much for your contribution, this is a very good PR!

File: crates/svm-builds/build.rs (hunk 1 above)
Comment (DaniPopes) [reply crates/svm-builds/build.rs:1]: This PR is rebased on top of #78

Comment (roynalnaruto) [reply crates/svm-builds/build.rs:1]: yup, just saw that 👍

**Comment by DaniPopes** [comment]
> Given the fact that there are several changes in the Cargo.toml files, I would prefer using a regular expression for allowing patch updates instead of specifying a patch.

> So something like ^1.0.0 ~1.0 instead of 1.0.17.

@roynalnaruto If you want to specify a patch later on you can just set the version when you need to, + you already maintain a Cargo.lock file so binary builds will be fine either way.

**Review by mattsse — APPROVED** [review]

**Review by Evalir — APPROVED** [review]
ack on change for future reference, lgtm
